### PR TITLE
setup new project

### PR DESCRIPTION
**Pull request by icerove**
setup for new mixpanel wallet project

### CODE REVIEW

**Review by vgrichina — COMMENTED** [review]
@icerove why is this done? Do we want to discard historical data?

File: src/middleware/index.js [hunk 1]
```diff
@@ -48,15 +49,64 @@ if (process.env.MIXPANEL_TOKEN) {
...
+        mixpanel.register({
+            network_id: networkId,
+            timestamp: new Date(),
+            account_id: wallet.accountId
+        })
+        mixpanel.people.set_once({
+            network_id: networkId,
+            stake: 0
+        })
+        if(trackingEvents.includes(action.type)){
```
Comment (vgrichina) [inline src/middleware/index.js:76]: nit: please format `if (` in consistent way with rest of the file

File: src/middleware/index.js [hunk 2]
```diff
@@ -48,15 +49,64 @@ if (process.env.MIXPANEL_TOKEN) {
 }
 
 const analyticsMiddleware = store => next => action => {
+    let createEvents = ['CREATE_NEW_ACCOUNT', 'CREATE_ACCOUNT_WITH_SEED_PHRASE','CHECK_ACCOUNT_AVAILABLE', 'CHECK_NEW_ACCOUNT' ]
+    let twoFAEvents = ['GET_2FA_METHOD', 'INIT_TWO_FACTOR', 'VERIFY_TWO_FACTOR', 'PROMPT_TWO_FACTOR', 'RESEND_TWO_FACTOR']
```
Comment (vgrichina) [inline src/middleware/index.js:53]: nit: these seem to be unused, is there plan to use them in upcoming PR?

File: src/middleware/index.js [hunk 3]
```diff
@@ -48,15 +49,64 @@ if (process.env.MIXPANEL_TOKEN) {
...
+    let ledgerEvents = ['GET_LEDGER_KEY', 'GET_LEDGER_PUBLIC_KEY', 'ADD_LEDGER_ACCESS_KEY', 'CONNECT_LEDGER', 'DISABLE_LEDGER',
+'GET_LEDGER_ACCOUNT_IDS', 'ADD_LEDGER_ACCOUNT_ID', 'SAVE_AND_SELECT_LEDGER_ACCOUNTS']
+    let accesskeyEvents = ['GET_ACCESS_KEYS', 'REMOVE_ACCESS_KEY','REMOVE_NON_LEDGER_ACCESS_KEYS', 'ADD_ACCESS_KEY', 'ADD_ACCESS_KEY_SEED_PHRASE']
+    let recoveryEvents = ['DELETE_RECOVERY_METHOD', 'RECOVER_ACCOUNT_SEED_PHRASE', 'SETUP_RECOVERY_MESSAGE_NEW_ACCOUNT']
+    let transactionEvents = ['DEPLOY_MULTISIG', 'CHECK_NEAR_DROP_BALANCE', 'SET_LEDGER_TX_SIGNED','SIGN_AND_SEND_TRANSACTIONS', 'SEND_MONEY', 'SWITCH_ACCOUNT']
+    let stakeEvents = ['UPDATE_STAKING', 'STAKE', 'UNSTAKE','WITHDRAW']
+    let trackingEvents = createEvents.concat(twoFAEvents).concat(ledgerEvents).concat(accesskeyEvents).concat(recoveryEvents).concat(transactionEvents).concat(stakeEvents)
```
Comment (icerove) [inline src/middleware/index.js:60]: @vgrichina  These events are all what we need and see this line, I include all events.

**Comment by icerove** [comment]
who will give approve and then merge?

File: src/middleware/index.js (hunk 3 above)
Comment (mattlockyer) [reply src/middleware/index.js:60]: @icerove there are no arguments from these events that will get sent to mixpanel correct?

e.g. RECOVER_ACCOUNT_SEED_PHRASE should NOT send any information about seed phrase to mixpanel.

**Comment by Patrick1904** [comment]
@vgrichina @kcole16 @mattlockyer this looks good to me. @vgrichina are we ok merging this?

**Comment by vgrichina** [comment]
@Patrick1904 @icerove I do not understand why are we changing mixpanel token. Is it ok to discard historical data already collected in mixpanel? Why do we want to start from scratch?

File: src/middleware/index.js (hunk 3 above)
Comment (icerove) [reply src/middleware/index.js:60]: this will not seed any extra information to mixpanel unless write down

**Comment by icerove** [comment]
@vgrichina Previous project has too much usefulness events and are mixed up with near-cli. We are okay to start from scratch since old one we cannot make any analysis there since there is not property in every tracking

**Comment by icerove** [comment]
The old data is still there in the old project but we need new one to make them all clean up

**Pull request merged**